can/simulator/rtos_main.cpp

```diff
@@ -0,0 +1,94 @@
+#include <iostream>
+#include <variant>
+
+#include "FreeRTOS.h"
+#include "can/core/can_message_buffer.hpp"
+#include "can/core/dispatch.hpp"
+#include "can/core/freertos_can_dispatch.hpp"
+#include "can/core/messages.hpp"
+#include "common/core/bit_utils.hpp"
+#include "common/core/freertos_message_buffer.hpp"
+#include "common/core/freertos_task.hpp"
+
+using namespace freertos_message_buffer;
+using namespace can_dispatch;
+using namespace can_message_buffer;
+using namespace can_messages;
+using namespace freertos_can_dispatch;
+using namespace freertos_task;
+
+struct HandlerA {
+    void handle(std::variant<std::monostate, GetSpeedRequest> &m) {
+        std::visit([this](auto o) { this->visit(o); }, m);
+    }
+
+    void visit(std::monostate &m) {
+        std::cout << "HandlerA monostate" << std::endl;
+    }
+
+    void visit(GetSpeedRequest &m) {
+        std::cout << "HandlerA get speed request" << std::endl;
+    }
+};
+
+struct HandlerB {
+    void handle(std::variant<std::monostate, StopRequest> &m) {
+        std::visit([this](auto o) { this->visit(o); }, m);
+    }
+
+    void visit(std::monostate &m) {
+        std::cout << "HandlerB monostate" << std::endl;
+    }
+
+    void visit(StopRequest &m) { std::cout << "HandlerB stop" << std::endl; }
+};
+
+// Handlers for the parsed messages
+static auto a_handler = HandlerA{};
+static auto b_handler = HandlerB{};
+
+// The dispatch targets.
+static auto a = freertos_can_dispatch::FreeRTOSCanDispatcherTarget<
+    256, HandlerA, GetSpeedRequest>{a_handler};
+static auto b =
+    freertos_can_dispatch::FreeRTOSCanDispatcherTarget<256, HandlerB,
+                                                       StopRequest>{b_handler};
+
+// Buffer poller tasks
+static auto dispatcher_task_a =
+    FreeRTOSTask<256, 5, decltype(a.poller)>("dispatcher_a", a.poller);
+static auto dispatcher_task_b =
+    FreeRTOSTask<256, 5, decltype(b.poller)>("dispatcher_b", b.poller);
+
+static auto dispatcher = Dispatcher(&a.buffer_target, &b.buffer_target);
+
+static auto buffer = FreeRTOMessageBuffer<1024>{};
+static auto poller =
+    freertos_can_dispatch::FreeRTOSCanBufferPoller(buffer, dispatcher);
```

Comment:
Now build the message buffer that simulates the message coming from the bus and connect it to the dispatcher.